### PR TITLE
If release_channel is active on the module, set min_master_version to…

### LINKED ISSUE #383: min_master_version and release_channel conflicting
I want to use the `release_channel` feature in 6.1.1, specifically in the `beta-private-cluster` module, however, skimming the code I'm pretty sure this applies to any of the `beta_*` modules.  In this specific case, I'm using the `release_channel` and specifying a `region` with 3 `zones`, resulting in a `master_version` (as of last Thursday) of 

```
master_version              = "1.14.8-gke.12"
```

The cluster deployed just fine. However now when I trying to apply an unrelated change, I'm seeing

```
        master_version              = "1.14.8-gke.12"
      ~ min_master_version          = "1.14.8-gke.17" -> "1.15.4-gke.22"
```

Trying to apply the change to the cluster results in the following error:

```
Error: googleapi: Error 400: Current master version cannot be updated when releaseChannel REGULAR is set., badRequest
```

I initially tried to use `ignore_changes` in my module invocation to ignore `min_master_version`, and immediately ran into the fact this can't currently be done.

### PR DESCRIPTION
**Pull request by kunalkg11**
Fixes #383 

Changes in the PR:

1. If release_channel is active on the module, set min_master_version to null
2. Modified beta-cluster example, removed zonal test and added regional test
3. Added release_channel to simple-regional-beta example

### CODE REVIEW

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -351,7 +351,7 @@ resource "google_container_node_pool" "pools" {
 
   cluster = google_container_cluster.primary.name
 
-  version = lookup(each.value, "auto_upgrade", false) ? "" : lookup(
+  version = lookup(each.value, "auto_upgrade", true) ? "" : lookup(
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl]: I think this was unintended

File: autogen/main/cluster.tf.tmpl [hunk 2]
```diff
@@ -56,7 +56,7 @@ resource "google_container_cluster" "primary" {
 {% endif %}
 
   subnetwork         = data.google_compute_subnetwork.gke_subnetwork.self_link
-  min_master_version = local.master_version
+  min_master_version = var.release_channel != null ? null : local.master_version
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl]: release_channel is beta so we would need to modify it such that it is not introduced in non beta modules

Comment (morgante) [reply autogen/main/cluster.tf.tmpl]: Correct. @kunalkg11 You'll need to make this conditional.

File: autogen/main/cluster.tf.tmpl (hunk 1 above)
Comment (kunalkg11) [reply autogen/main/cluster.tf.tmpl]: Fixed.

File: autogen/main/cluster.tf.tmpl (hunk 2 above)
Comment (kunalkg11) [reply autogen/main/cluster.tf.tmpl]: Fixed.

**Pull request merged**